google/cloud/internal/timer_queue_test.cc

```diff
@@ -76,27 +78,58 @@ TEST(TimerQueueTest, CancelTimers) {
   for (auto& t : runners) t.join();
 }
 
-TEST(TimerQueueTest, ScheduleEarlierTimerSingleRunner) {
+/// @test Verify timers are executed in order.
+TEST(TimerQueueTest, SingleRunnerOrdering) {
+  // Populate some delays with intentionally out-of-order and in-order data.
+  using duration = std::chrono::system_clock::duration;
+  std::vector<duration> const delays{
+      duration(10), duration(9),  duration(8), duration(5), duration(20),
+      duration(21), duration(22), duration(4), duration(3),
+  };
+
+  // Create the timer queue, but do not run the servicing thread until the
+  // timers are scheduled. Otherwise "later" timers may be immediately expired
+  // because clocks are hard.
   auto tq = TimerQueue::Create();
+
+  auto const start = std::chrono::system_clock::now();
+  std::vector<std::chrono::system_clock::time_point> expected(delays.size());
+  std::transform(delays.begin(), delays.end(), expected.begin(),
+                 [start](auto d) { return start + d; });
+  // We are going to store the expiration of each timer in this vector, we do
+  // not need to protect it with a mutex because only one thread operates on
+  // it at a time.
+  std::vector<std::chrono::system_clock::time_point> actual;
+  std::vector<future<Status>> timers(expected.size());
+  std::transform(expected.begin(), expected.end(), timers.begin(), [&](auto d) {
+    // As the timers expire we add their time (if successful) to `expirations`,
+    // and return the `Status`.
+    return tq->Schedule(d).then([&, d](auto f) {
+      auto e = f.get();
+      if (!e) return std::move(e).status();
+      std::ostringstream os;
+      os << "mismatched " << d << " != " << *e;
```

Comment:
Move this into the actual `*e != d` block?

Reply:
Done